pkg/graveler/committed/diff_test.go

```diff
@@ -238,6 +275,98 @@ func TestDiffCancelContext(t *testing.T) {
 		t.Fatalf("Err() returned %v, should return context.Canceled", err)
 	}
 }
+func TestNextRange(t *testing.T) {
+	ctx := context.Background()
+	it := committed.NewDiffIterator(ctx,
+		newFakeMetaRangeIterator([][]string{{"k1", "k2"}, {"k3", "k4"}}, [][]string{{"i1", "i2"}, {"i3", "i4"}}),
+		newFakeMetaRangeIterator([][]string{{"k3", "k4"}, {"k5", "k6"}}, [][]string{{"i3a", "i4a"}, {"i5", "i6"}}))
+
+	t.Run("next range - in middle of range", func(t *testing.T) {
+		if !it.Next() { // move to range k1-k2
+			t.Fatal("expected iterator to have value")
+		}
+		record, rng := it.Value()
+		if record != nil {
+			t.Fatal("expected record to be nil")
+		}
+		if !it.Next() { // move to k1
+			t.Fatal("expected iterator to have value")
+		}
+		record, rng = it.Value()
+		if record == nil || string(record.Key) != "k1" {
+			t.Fatal("expected record key to be k1")
+		}
+		if !it.NextRange() { // move to k3
```

Comment:
`NextRange` enters the range?  I think that's different from the single iterator (see [documentation](https://github.com/treeverse/lakeFS/blob/303b949346864bf84ce3f8e6675487b5d96e06a8/pkg/graveler/committed/meta_range.go#L17)).  So at the very least we must document this function.  Ideally we should also know why they are different.

Reply:
It will not enter the range if there is a range, in this case k3 is part of two different ranges.
Basically the next range is a "headerless" range. 
Adding a comment